### PR TITLE
feat(plugins): add plugin stage test

### PR DESCRIPTION
**Pull request by danielpeach**

### CODE REVIEW

File: dev/validate-config-template.yml
Comment (danielpeach) [inline dev/validate-config-template.yml:235]: This plugin version hasn't been released (will do that in a second) but will allow Orca to load the plugin without any additional extension-specific config (there aren't any `hal` commands for that config).

`Armory.RandomWaitPlugin@1.1.x` won't load in Spinnaker 1.19.x but will load in `master-latest-unvalidated`. I don't see any config options for only running against specific minor versions / the nightly build. Assuming that's somehow set on the build server itself but please let me know if I'm missing something.

**Comment by danielpeach** [comment]
@ttomsu or @maggieneterval or @ezimanyi please review!

File: dev/validate-config-template.yml
Comment (maggieneterval) [reply dev/validate-config-template.yml:235]: I think unless we cherry-pick this change back to the 1.19/1.18/1.17 branches, it will only run as part of the master build, so you should be good there.

Would we want to test that this passes by running a flow off of this branch before merging? (Or maybe you already did that?)

**Comment by ttomsu** [comment]
Given the amount of broken and failing builds we've had recently, I'd like to hold off on this until the next release branch is cut, unless @danielpeach plugins are going to be launched during the next release?

File: dev/validate_bom__config.py
Comment (ezimanyi) [inline dev/validate_bom__config.py:1679]: One thing that worries me a bit here is that we're loading the plugins for *all* tests, so if there's some issue with that particular plugin or if it breaks in the future the deploy will fail as opposed to just the plugin test.

That being said, by design we only deploy once as it takes a long time; and I guess the version is pinned, right, so it won't randomly break?

**Review by ezimanyi — APPROVED** [review]
Seems reasonable to me, just one question!

File: dev/validate-config-template.yml
Comment (danielpeach) [reply dev/validate-config-template.yml:235]: I've confirmed locally that it spits out the right hal commands and that the test passes against my Spinnaker installation. 

Is there a way for me to test against the Jenkins you have running? Would be really great if that were possible without becoming a member of the build cops group... 😄

File: dev/validate_bom__config.py
Comment (danielpeach) [reply dev/validate_bom__config.py:1679]: That's a fair thing to worry about and probably worth thinking about. This plugin is very narrowly scoped but other plugins might have implications for other integration tests (e.g., a custom clouddriver credential loader) or even implications for other plugins. This probably isn't the right way to test individual plugins.

Yes re: the plugin version being pinned. Loading plugins could break the whole test suite if there's some bug in the Kork plugin-loader code that prevents a service from starting up. But that kind of bug exists for any feature that's enabled that happens to bring down a Spinnaker service...

Comment (ezimanyi) [reply dev/validate_bom__config.py:1679]: Yeah, I think this makes sense then given that it's a super simple pinned plugin that is just there to ensure that we can deploy with a plugin.

**Comment by danielpeach** [comment]
@ttomsu OK to merge now that the release branch has been cut?

**Pull request merged**
